packages/KCardCatalog/KCatalogItem.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <router-link
+    v-if="locationParam"
+    :to="locationParam">
+    <KCard
+      :data-testid="item ? `${item.title.replace(' ', '-')}-catalog-item` : 'catalog-item'"
+      :test-mode="testMode"
+      class="grid-item d-flex flex-column overflow-hidden k-card-catalog-item"
+      has-hover
+      role="button"
+      tabindex="0"
+      @click.native="handleCardClick($event, item)"
+    >
+      <template v-slot:title>
+        <slot name="cardTitle">
+          {{ item.title }}
+        </slot>
+      </template>
+      <template v-slot:body>
+        <div
+          :class="{ 'multi-line-truncate': truncate }">
+          <slot name="cardBody">
+            {{ item.description }}
+          </slot>
+        </div>
+      </template>
+    </KCard>
+  </router-link>
```

Comment:
In an effort to reduce the amount of duplicated code, would it be possible to conditionally wrap the KCatalogItem with a router-link in the KCardCatalog component instead of here?

I think it could look something like this:
```vue
<template v-for="item in items">
  <router-link v-if="item.locationParam" :key="item.key ? item.key : null">
    <KCatalogItem
      :item="item"
      :location-param="item.locationParam ? item.locationParam : ''"
      :truncate="!noTruncation"
      :test-mode="testMode"
      class="catalog-item"
    />
  </router-link>
  <KCatalogItem
    v-else
    :key="item.key ? item.key : null"
    :item="item"
    :truncate="!noTruncation"
    :test-mode="testMode"
    class="catalog-item"
  />
</template>
```